client/rpc/src/lib.rs

```diff
@@ -272,6 +297,75 @@ where
         Ok(read_fs.root())
     }
 
+    async fn is_file_in_forest(&self, key: Option<String>, file_key: H256) -> RpcResult<bool> {
+        let key = FSH::Key::from(key.unwrap_or_default());
+
+        let fs =
+            self.forest_storage_handler.get(&key).await.ok_or_else(|| {
+                into_rpc_error(format!("Forest storage not found for key {:?}", key))
+            })?;
+
+        let read_fs = fs.read().await;
+        Ok(read_fs
+            .contains_file_key(&file_key)
+            .map_err(into_rpc_error)?)
+    }
+
+    async fn is_file_in_file_storage(&self, file_key: H256) -> RpcResult<bool> {
+        // Acquire FileStorage read lock.
+        let read_file_storage = self.file_storage.read().await;
+
+        // See if the file metadata is in the File Storage.
+        match read_file_storage
+            .get_metadata(&file_key)
+            .map_err(into_rpc_error)?
+        {
+            None => return Ok(false),
+            Some(_) => Ok(true),
+        }
+    }
+
+    // Note: this method could use either the file storage or the forest storage, but it's using the forest storage.
+    // WARNING: Right now, forests don't have the file metadata saved to them, so don't expect to get the file
```

Comment:
We could also add a fallback to either get it from forest or file manager.

Reply:
Do you think we should have two methods? One from the Forest storage and one from the File storage?
If it helps I believe we don't really need access to metadata that's only on the File storage, since for now the idea is to only use it when deleting already-stored files from an insolvent user (which should be in the Forest storage). Although now that I think about it, we also have storage requests that might be pending...